### DISCUSSION TITLE
Add metadata column to meta()

### DISCUSSION THREAD

**Discussion by riazarbi** [ropensci/targets#482] [Ideas]
Hello,

I think it would be nice to be able to add a text description to a target. Bonus points if you can add a character vector, but not sure if the `meta` text-based backend can handle it. 

What I’d like to be able to do is specify

```
 tar_target(tar1,
                   function(arg1, arg2, arg3),
                   …,
                  annotations = c(“class” = “geospatial object”,
                                              “description” = “beer bler bler”))
```

And then for `tar_meta()$annotations[1]` to give me a named vector of the form:

```
c(“class” = “geospatial object”, “description” = “beer bler bler”)
```

I see this as being an obvious enhancement, as it allows me to embed metadata about targets into `meta()`, which then allows me to programmatically generate documentation along with a workflow. 

However, it could be some sort of anti-pattern that you are explicitly trying to avoid. 

Thoughts?

**Comment by riazarbi** [comment]
Oops, I meant to ask this in the `targets` repo. Shall I delete it and repost it there?

**Comment by wlandau** [comment]
This was also asked in #467. I think literate programming is a more complete solution to the underlying need. You can either create [literate programming targets](https://books.ropensci.org/targets/files.html#literate-programming) as part of a pipeline, or use [Target Markdown](https://books.ropensci.org/targets/markdown.html) (in development `targets`) which is a `knitr`-focused way to set up one or more pipelines. Both options let you narrate the pipeline and the output in lengthy prose.

**Comment by riazarbi** [comment]
Thanks @wlandau, I had not seen the new targets markdown format. I prefer plain R scripts, but will give it a shot. 

This library is developing so fast, it often seems you've implemented something a week before I needed it. I've started watching the repo now to keep track!
